lib/matplotlib/axes/_axes.py

```diff
@@ -2025,6 +2025,8 @@ def make_iterable(x):
                 bottom = [0]
 
             nbars = len(left)
+            if len(height) == 1:
```

Comment:
These 6 lines of code could be factored out of the if statement. Only `nbars` need to be inside the if - elif statement.